Fixed foreground, Update foundation to 5.5.3

paladox opened the pull request:
This fixes the skin so it is usable with foundation 5 now.

Fixed drop down so it works now.

We should prepare thingles:feature/foundation5 branch to be merged into develop branch once we get everything working. @Hutchy68 Could we should go through this branche finding anything that doesn't work and fixing it.

Action button needs to be made smaller.

It seems to be very stable now. Works really good.

With feedback from @Hutchy68 also some code was created by @Hutchy68

Hutchy68 commented on Foreground.skin.php:
```diff
@@ -199,11 +199,11 @@ public function execute() {
 				<div id="p-cactions" class="large-12 columns">
 					<?php if ($wgUser->isLoggedIn() || $wgForegroundFeatures['showActionsForAnon']): ?>
-						<a href="#" data-dropdown="drop1" class="button dropdown small secondary radius"><i class="fa fa-cog"><span class="show-for-medium-up">&nbsp;<?php echo wfMessage( 'actions' )->text() ?></span></i></a>
+						<a href="#" class="button dropdown small secondary radius"><i class="fa fa-cog"><span class="show-for-medium-up">&nbsp;<?php echo wfMessage( 'actions' )->text() ?></span></i></a>
```
Missing the `data-options=` which is what caused issues recently with 1.2. This is what I use on a production wiki running 1.2 with Foundation 4. This should carry forward to 5 without any problems and you can keep the classes you tweaked here. Those are just style.

```
<a href="#" data-dropdown="hover1" data-options="is_hover:true; ignore_repositioning;" class="button dropdown small secondary radius"><i class="fa fa-cog"><span class="show-for-medium-up">&nbsp;<?php echo wfMessage( 'actions' )->text() ?></span></i></a>
	<!--RTL -->
	<ul id="hover1" data-dropdown-content="content" aria-hidden="true" class="views large-12 columns right f-dropdown">
		<?php foreach( $this->data['content_actions'] as $key => $item ) { echo preg_replace(array('/\sprimary="1"/','/\scontext="[a-z]+"/','/\srel="archives"/'),'',$this->makeListItem($key, $item)); } ?>
		<?php wfRunHooks( 'SkinTemplateToolboxEnd', array( &$this, true ) );  ?>
	</ul>
```

paladox replied:
Ok You also have to add hover_timeout:5000; or it will close as soon as you move the mouse. But when you try to close it by hovering out of the box there is a 5sec delay how can we automate this so it dose this only if we leave the box.

paladox replied:
Ive found a way by doing hover_timeout:100; instead. Also the action button seems to be fixed in foundation5 since followed your suggestion and it fixed the problem. And I also did some minor changes and seems js is working much better more stable and improved performance since it is loading less modules.

Hutchy68 commented on assets/scripts/foreground.js:
```diff
@@ -1,77 +1,78 @@
-  jQuery('ul#drop1').mouseleave(function(){
-    jQuery('ul#drop1').removeClass('open').css('top', '-9999px').css('left', '785px');
-  });
+	// Make the Page Action button respond to hover
```
If you use native Foundation on hover, this will cause issues and breaks the on hover of the action button.

Hutchy68 commented on skin.json:
```diff
@@ -57,7 +61,12 @@
 				"assets/scripts/foundation/foundation.alert.js",
 				"assets/scripts/foundation/foundation.accordion.js",
 				"assets/scripts/foundation/foundation.abide.js",
-				"foreground/assets/scripts/foreground.js"
```
I think we need to have a discussing about how many of the Foundation js functions to load. Things like abide, Magellan, Sliders, etc. If the skin isn't using them, can the user use them on pages? Magellan is cool, but how can a user use it on their wiki page? Most require markup which will be sanitized unless the wiki admin allows raw html.

paladox replied:
I went ahead and removed those three since they are unused by this repo and cause extra server load. I am fiding a tiny bit speed bump removing them.

paladox commented:
Ok I have fixed the problems. Js now works properly. Ive fixed the searchbar. Dropdown works properly now. The only thing is the little arrow that is pointing up in the action drop down menu it is on the left where as it should be on the right.

Hutchy68 commented:
Getting there, search bar is still off but I'm more concerned about this...
![warning](https://cloud.githubusercontent.com/assets/3527469/11914164/4e29d64c-a646-11e5-9a9b-8ccd79996e06.PNG)

Check your logs or allow error reporting output

paladox commented:
Oh I'm not getting that.

What mw version are you using. I'm using mw 1.27.

Hutchy68 commented:
1.26

paladox commented:
Ok. I am not getting the error your getting are you sure you applied this whole patch correctly.

Hutchy68 commented:
Yeah and I even downloaded a zip of your patch-27, exploded and installed that clean to double check.

Hutchy68 commented:
Do you the tabs feature set to true

paladox commented:
Ok. Does it work without this patch. If it does then we know that it is a problem with this patch. But requiring position was made a required type in 1.27 so maybe 1.26 carn't cope with that so it errors that out.

paladox commented:
Yes I do have tabs set to true.

paladox commented:
It seems that useForegroundTabs isen't present in develop branch. Do you know if that was removed if so it should be done here too.

paladox commented:
@Hutchy68 Please try this new patch I removed the tabs features since not sure what it was for.

Hutchy68 commented:
It was only needed for the 5 change of how tabs are constructed

paladox commented:
Oh seems the tabs look good without this. What do you mean by tabs. do you mean the ones on top bar.

Hutchy68 commented:
Setting to false works, but tabs is being used by some. Example, Wikiapiary needs it.

paladox commented:
Ok well we could set it false by default and wikiapriary would then need to set it true there.

paladox commented:
But I thought this setting isen't in the develop branch. Nor is the tabs code. It was added in foundation5 to fix tabs.

Hutchy68 commented:
http://foundation.zurb.com/sites/docs/v/5.5.3/components/tabs.html

But that doesn't fix the warning

Hutchy68 commented:
It's not in develop because that's 4

paladox commented:
Oh ok.

Hutchy68 commented:
I think the problem is we probably need to consider extending the class of output page instead of using a switch / case.

paladox commented:
Oh ok.

paladox commented:
Try this patch I just uploaded please.

paladox commented:
If we need to do extending the class of output page  could I have help to do that.

Hutchy68 commented:
I'll check it when I get back home. On mobile now.

paladox commented:
ok thanks.

paladox commented:
@Hutchy68 Please could you try the new patch ive uploaded. I think the problem was we were duplicating the switch for that specific code. So I removed the duplicate and moved it to the switch that was using it.

paladox commented:
@Hutchy68 could you do a review off this patch and identify any issues or suggestions you have please.

Hutchy68 commented:
I'll check it

paladox commented:
Thanks.

Hutchy68 commented:
Still getting errors from tabs feature. ob_flush. Search button is out of line and offset below form. Dropdown is still an issue for actions. 

I am glad you put in the indicator.  I was going to start a discussion in an issue after 2.0 is near release. I think it's time we start considering placement of the echo icon along with the new indicator messages now available in 1.26 and message notifications from flow message notifications from flow extension. We could even move the print page icon w/link. There would be too much to the left of the dropdown actions button w/ alerts, messages and indicators but a small div floated right directly above actions, tweak margins at top, and make the font-size .9rem looks fine. I have been experimenting with a div. 

I think we are at the point where it might be better to just abandon the f5 feature branch and merge in the clean f5 update into develop. Comparing, that branch is a mess and I don't want to see you waste your time on it. It is over a year old and really was one of the early tries at f5.

paladox commented:
Ok. So you would like me to open a pull and start merging into develop patch ok I will. Could this pull be left open since I made some changes and want to compare to get them back to normal. With tabs problem could you test with mediawiki 1.23, 1.24, 1.25 please if they work I think with tabs we would need to stop tabs code from being used in 1.26 until the problem is fixed. I would need help to fix the other issues you spotted please. I am not sure how to fix the search bar since it looks ok too me so I am missing something your not.

paladox commented:
@Hutchy68 do you get the tabs error in 1.27 alpha if so then it's the way I've setup my wiki.

paladox commented:
@Hutchy68  I haven't tested the indicators though but the help button looks like an indicator so yes we need a new placement possibly to the right instead of left. 

With the echo button what do you suggest.

I haven't used flow so would end know where to place the icon.

Also thank you for reviewing this.

Also reviewing you patch you uploaded was really good too just some feedback which I could give but not sure which patch we should do.

paladox commented:
I'm not sure how to add your name to the person that help create this.

paladox commented:
Maybe about tbs I could ask on the mediawiki irc channel and ask why it does it and what should I do to fix. Could it be because it is done after headers when it should be done before.

paladox commented:
I found information on the header problem at https://www.mediawiki.org/wiki/Manual:Errors_and_symptoms#Warning:_Cannot_modify_header_information_-_headers_already_sent_by_.28....29

Hutchy68 commented:
No, not a BOM file. If you don't have output_buffering = on and use ob_start() after the headers are sent then you get an error when you ob_clean... or flush. If you need to use the buffering you must set the ob_start() at the very top of the file right after `<?php` so the buffering kicks in before anything is sent otherwise php chunks the response(i.e. will send the headers) before everything is executed by php.

Hutchy68 commented:
Not even sure why they used output buffering. The skin controls the output of the body content and it can be modified before we output it on the fly.

paladox commented:
Should I remove ob_start and not sure what else to remove.

paladox commented:
@Hutchy68 Do you know off a way that we can increase the performance since it is really slow it is taking me 27 seconds to load a page which is slow.

paladox commented:
It seems we now have two help buttons one under navigation drop down and one under tools drop down.

Hutchy68 commented:
feature branch is being closed, too old and too many issues now fixed in develop branch
